### PR TITLE
Added openbabel and vina and basic tests.

### PR DESCRIPTION
**Pull request by dakotablair**
Added Open Babel 3.0.0 and AutoDock Vina 1.1.2 to the image. Additionally, tests using a sample receptor and sample ligands are included to demonstrate usage of each.

- [x] Update Dockerfile to install vina and obabel.
- [x] Write a test converts a PDB receptor from PDB into PDBQT via obabel and grep
- [x] Write a test converts a list of SDF ligands from PubChem into PDBQT via obabel
- [x] Write a test which takes in a receptor and list of ligands in PDBQT format and runs vina on them and saves the vina output into a simple report.

### CODE REVIEW

File: test/data/Structure2D_CID_139024764.sdf [hunk 1]
```diff
@@ -0,0 +1,369 @@
+139024764
+  -OEChem-06222210252D
```
Comment (imrFAMU) [inline test/data/Structure2D_CID_139024764.sdf:2]: I am assuming that this is the ligand file

File: test/unit_tests/test_kb_ad_vina_utils.py [hunk 2]
```diff
@@ -1,18 +1,125 @@
...
+        obabel_cmd_receptor,
+        shell=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    ) as proc:
+        proc.communicate()
+
+    with subprocess.Popen(
+        f"""grep -e "^\\(ATOM\\|TER\\)" {receptor}.obabel.pdbqt \\
+            >> {receptor}.obabel.clean.pdbqt
```
Comment (imrFAMU) [inline test/unit_tests/test_kb_ad_vina_utils.py:60]: I understand that we are matching all lines with ATOM or TER for the cleaning of the pdbqt, but what does grep -e do?

File: test/unit_tests/test_kb_ad_vina_utils.py [hunk 3]
```diff
@@ -1,18 +1,125 @@
...
+            logging.info(stdout.decode("utf-8", "ignore"))
+        with subprocess.Popen(
+            "head -n27 test.log | tail -n1 | awk '{ print $2 }'",
+            shell=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        ) as proc:
+            stdout, stderr = proc.communicate()
+            assert approximatelyEqual(
+                affinity[ligand],
```
Comment (imrFAMU) [inline test/unit_tests/test_kb_ad_vina_utils.py:122]: Not sure i fully understand the assert cmd with affinity[ligand]

File: test/unit_tests/test_kb_ad_vina_utils.py (hunk 2 above)
Comment (dakotablair) [reply test/unit_tests/test_kb_ad_vina_utils.py:60]: The `-e` tells `grep` to expect the pattern as the value for that parameter. I tend to think of it as less ambiguous.

File: test/unit_tests/test_kb_ad_vina_utils.py (hunk 3 above)
Comment (dakotablair) [reply test/unit_tests/test_kb_ad_vina_utils.py:122]: The `affinity` dictionary holds the output affinities we expect from AutoDock Vina. I would also expect by specifying a `seed` that we should get _exactly the same_ floating point value at the end, but in practice this does not happen which is what requires the use of `approximatelyEqual`. In the mean time I learned about `math.isclose` and at some point I will refactor the code to use this function instead.

**Review by imrFAMU — APPROVED** [review]

**Pull request merged**